cmd/deploy.go

```diff
@@ -315,6 +356,61 @@ For more options, run 'func deploy --help'`, err)
 	}
 	cmd.SetContext(cfg.WithValues(cmd.Context())) // Some optional settings are passed via context
 
+	// Validate cluster connection before building
+	if err = validateClusterConnection(); err != nil {
+		if errors.Is(err, fn.ErrInvalidKubeconfig) {
+			kubeconfigPath := os.Getenv("KUBECONFIG")
+			if kubeconfigPath == "" {
+				kubeconfigPath = "~/.kube/config (default)"
+			}
+
+			return fmt.Errorf(`%w
+
+The kubeconfig file at '%s' does not exist or is not accessible.
+
+Try this:
+  export KUBECONFIG=~/.kube/config           Use default kubeconfig
+  kubectl config view                        Verify current config
+  ls -la ~/.kube/config                      Check if config file exists
+
+For more options, run 'func deploy --help'`, fn.ErrInvalidKubeconfig, kubeconfigPath)
+		}
+
+		if errors.Is(err, fn.ErrClusterNotAccessible) {
+			errMsg := err.Error()
+
+			// Case 1: Empty/no cluster configuration in kubeconfig
+			if strings.Contains(errMsg, "no configuration has been provided") ||
+				strings.Contains(errMsg, "invalid configuration") {
+				return fmt.Errorf(`%w
+
+Cannot connect to Kubernetes cluster. No valid cluster configuration found.
+
+Try this:
+  minikube start                             Start Minikube cluster
+  kind create cluster                        Start Kind cluster
+  kubectl cluster-info                       Verify cluster is running
+  kubectl config get-contexts                List available contexts
+
+For more options, run 'func deploy --help'`, fn.ErrClusterNotAccessible)
+			}
+
+			// Case 2: Cluster is down, network issues, auth errors, etc
+			return fmt.Errorf(`%w
+
+Cannot connect to Kubernetes cluster.
+
+Try this:
+  kubectl cluster-info                       Verify cluster is accessible
+  minikube status                            Check Minikube cluster status
+  kubectl get nodes                          Test cluster connection
+
+For more options, run 'func deploy --help'`, fn.ErrClusterNotAccessible)
```

Comment:
i would be in favor of making these into functions so they dont take up so much space in the main-flow function